### PR TITLE
cmd/geth: add banner for devmode on console

### LINKED ISSUE #18243: Banner on the console
Print a banner on the console with general information. If `--dev` was specified, add "welcome to power user mode".

### PR DESCRIPTION
**Pull request by nedifi**
fixes #18243

> You are running Geth in --dev mode. Please note the following:
>      1.  This mode is only intended for fast, iterative development
without assumptions on security or persistence.
>     2.  The database is created in memory unless specified otherwise. Therefore, shutting down your computer or losing power will wipe your entire block data and chain state for your dev environment.
 >     3.  A random, pre-allocated developer account will be available and
unlocked as eth.coinbase, which can be used for testing. The
random dev account is temporary, stored on a ramdisk, and will
be lost if your machine is restarted.
>      4.  Mining is enabled by default. However, the client will only seal blocks if transactions are pending in the mempool. The miner's minimum accepted gas price is 1.
>      5.  Networking is disabled; there is no listen-address, the maximum number of peers is set to 0, and discovery is disabled.

### CODE REVIEW

**Review by rjl493456442 — APPROVED** [review]
Thanks, lgtm

File: cmd/geth/main.go [hunk 1]
```diff
@@ -463,3 +463,33 @@ func unlockAccounts(ctx *cli.Context, stack *node.Node) {
...
+                your entire block data and chain state for your dev environment.
+            3.  A random, pre-allocated developer account will be available and
+                unlocked as eth.coinbase, which can be used for testing. The
+                random dev account is temporary, stored on a ramdisk, and will
+                be lost if your machine is restarted.
+            4.  Mining is enabled by default. However, the client will only seal
+                blocks if transactions are pending in the mempool. The miner's
+                minimum accepted gas price is 1.
+            5.  Networking is disabled; there is no listen-address, the maximum
+                number of peers is set to 0, and discovery is disabled.`
```
Comment (rjl493456442) [inline cmd/geth/main.go:493]: Just nitpick that could you add one more blank line at the end? So that we can split banner and geth log clearly.

**Comment by holiman** [comment]
I'm personally not super-fond of the ascii art, would rather just omit that

**Comment by nedifi** [comment]
I'm happy with or without ASCII art. The main contribution of this PR is explaining what `--dev` actually implies.

Now, without the banner, I could also wrap each line in a `log.Warn()` to be more consistent with the logging format.

**Comment by fjl** [comment]
I've changed it back to use a single message, printing multiple just looked weird.

**Pull request merged**